chain/chunks/src/lib.rs

```diff
@@ -690,6 +706,17 @@ impl ShardsManager {
                     shard_id,
                     &chunk_hash,
                     false,
+                    // Request from a single node for missing chunks in old blocks
+                    // This is to save the number of requests a node sends
+                    // For old blocks, we can be reasonably sure that some peer has downloaded
+                    // the full chunk and we can get the chunk from them
+                    // TODO: when we enable challenges, this may break data availability
+                    //    if not many nodes receive the chunk before a block gets old (no longer the head)
+                    //    We leave this issue here for now because it is not a problem now when
+                    //    each shard is tracked by the majority of nodes in the network
+                    //    and we may change the way how chunks are distributed to non-validators
+                    //    https://github.com/near/nearcore/issues/3890
+                    old_block,
```

Comment:
Are we confident that this is indeed a better approach? The issue I see here is that if the node itself is a validator and it is slightly behind (>1 block), this will mean that the node will request parts from a single source, which may be less optimal than requesting from multiple validators. On the other hand, if the node is very behind, or that it is not a validator, then it probably doesn't matter that much. In short I don't think `old_block` captures what we really want here

Reply:
I thought about making the condition `old_block && non_validator`, the issue is that if a validator is behind, then it will request missing chunks for multiple blocks. If all of these requests are sent to part owners, that seems like a lot of requests.

I am still trying to figure out why the OOM occurred for testnet canary. Maybe it is not related to sending too many chunk requests. Then I agree we should restrict the change to only `non-validators`.